### DISCUSSION TITLE
Allowing Lambda Expressions with Statement Bodies to be Converted to Expression Trees

### DISCUSSION THREAD

**Discussion by AdamSobieski** [dotnet/csharplang#3093] [Language Ideas]
I would like to open an issue for discussing allowing lambda expressions with statement bodies to be converted to expression trees in C#.

Pertinent to this issue is [CS0834](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/cs0834) (a lambda expression with a statement body cannot be converted to an expression tree).

**Comment by Joe4evr** [comment]
This falls under #158.
